base/broadcast.jl

```diff
@@ -15,6 +15,13 @@ typealias ScalarType Union{Type{Any}, Type{Nullable}}
 # fallbacks for some special cases
 @inline broadcast(f, x::Number...) = f(x...)
 @inline broadcast{N}(f, t::NTuple{N,Any}, ts::Vararg{NTuple{N,Any}}) = map(f, t, ts...)
+function broadcast!{T,S,N}(::typeof(identity), x::Array{T,N}, y::Array{S,N})
+    if size(x) == size(y)
+        copy!(x, y)
+    else
+        Broadcast.broadcast_c!(identity, Array, Array, x, y)
```

Comment:
This definition living within the `Broadcast` module, no need for the qualification?